### ISSUE TITLE
linter: no-unused-vars false positives when variable and type have same name

### ISSUE THREAD

**Issue by EYHN** [oxc-project/oxc#8236]
### What version of Oxlint are you using?

0.15.5

### What command did you run?

oxlint

### What does your `.oxlint.json` config file look like?

without config file

### What happened?

with typescript file:

```ts
// bad.ts
const a = {
  hello: "world",
};

export type a = typeof a;
```

it shows error when i run oxlint

```
  × eslint(no-unused-vars): Type alias 'a' is declared but never used. Unused type aliases should start with a '_'.
   ╭─[src\bad.ts:1:7]
 1 │ const a = {
   ·       ┬
   ·       ╰── 'a' is declared here
 2 │   hello: "world",
   ╰────
  help: Consider removing this declaration.
```

But when I rewrite type a to type b

```ts
// good.ts
const a = {
  hello: "world",
};

export type b = typeof a;
```

no error will show

```
Found 0 warnings and 0 errors.
```

It seems that oxlint cannot correctly handle the situation where the variable and type with the same name. but this code can pass in eslint. so I think it's a bug

**Comment by Sysix** [comment]
this should be fixed in 0.15.7:

https://github.com/oxc-project/oxc/releases/tag/oxlint_v0.15.7
